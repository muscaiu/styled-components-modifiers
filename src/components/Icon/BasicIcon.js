import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import FontAwesome from 'react-fontawesome';
// import faStyles from 'font-awesome/css/font-awesome.css'
import { omit } from 'lodash';

function UnstyledIcon(props) {
    const iconProps = omit(props, ['modifiers']);
    console.log('iconProps:', iconProps)
    //return  <FontAwesome {...iconProps} />
    return (
        <FontAwesome
            className="super-crazy-colors"
            name="home"
            // cssModule={faStyles}
            // size="2x"
            // spin
            style={{ textShadow: '0 1px 0 rgba(0, 0, 0, 0.1)' }}
            {...iconProps}
        />
    )
}

const BasicIcon = styled(UnstyledIcon)`
  font-size: 1rem;
`;

BasicIcon.propTypes = {
    name: PropTypes.string.isRequired
};

export default BasicIcon;
