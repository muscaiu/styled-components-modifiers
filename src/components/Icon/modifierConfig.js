export default {
    left: () => `
    padding-right: 0.5rem;
  `,
    right: () => `
    padding-left: 0.5rem;
  `,
};