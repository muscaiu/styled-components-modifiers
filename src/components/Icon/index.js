import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';

import modifierConfig from './modifierConfig';
import BasicIcon from './BasicIcon';

const Icon = styled(BasicIcon)`
  ${applyStyleModifiers(modifierConfig)}
`;

export default Icon;
