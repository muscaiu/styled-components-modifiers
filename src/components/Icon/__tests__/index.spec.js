import React from 'react'
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import Icon from '../index';


describe('Icon Element', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <Icon name="home" />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('adds the left modifier', () => {
        const tree = renderer.create(
            <Icon name="home" modifiers={['left']} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('adds the right modifier', () => {
        const tree = renderer.create(
            <Icon name="home" modifiers={['right']} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
