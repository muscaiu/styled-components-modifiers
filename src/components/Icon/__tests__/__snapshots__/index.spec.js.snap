// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Element adds the left modifier 1`] = `
.c0 {
  font-size: 1rem;
  padding-right: 0.5rem;
}

<span
  aria-hidden={true}
  className="fa fa-home c0"
  modifiers={
    Array [
      "left",
    ]
  }
/>
`;

exports[`Icon Element adds the right modifier 1`] = `
.c0 {
  font-size: 1rem;
  padding-left: 0.5rem;
}

<span
  aria-hidden={true}
  className="fa fa-home c0"
  modifiers={
    Array [
      "right",
    ]
  }
/>
`;

exports[`Icon Element renders correctly 1`] = `
.c0 {
  font-size: 1rem;
}

<span
  aria-hidden={true}
  className="fa fa-home c0"
/>
`;
